blocks/graph.py

```diff
@@ -37,6 +37,7 @@ def _get_variables(self):
         self.applies = set()
         self.application_calls = set()
         self.updates = []
+        self.auxiliary_variables = set()
```

Comment:
Sorry for this monologue, but back to the first topic: I'll tag all auxiliary variables with `VariableRole.AUXILIARY`, that way the same method can be used to filter out `VariableRole.COST` and `VariableRole.AUXILIARY`. Makes more sense.